File: Java/lab14_Chain/task_2_3/src/Image.java

public class Image {

    public enum Type {
        bmp,
        png,
        jpg
    }

    private Type type;
    private String camera;
    private String creationDate;
    private String creationTime;
    private String name;

    public Type getType() {
        return type;
    }

    public void setType(Type type) {
        this.type = type;
    }

    public String getCamera() {
        return camera;
    }

    public void setCamera(String camera) {
        this.camera = camera;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(String creationDate) {
        this.creationDate = creationDate;
    }

    public String getCreationTime() {
        return creationTime;
    }

    public void setCreationTime(String creationTime) {
        this.creationTime = creationTime;
    }

    @Override
    public String toString() {
        return "Image{" +
                "type=" + type +
                ", camera='" + camera + '\'' +
                ", creationDate='" + creationDate + '\'' +
                ", creationTime='" + creationTime + '\'' +
                ", name='" + name + '\'' +
                '}';
    }
}

File: Cpp/lab14_Chain/task_2_3/src/image.cpp
#include "image.hpp"


Image::Image() {}

Image::Type Image::getType() const
{
  return this->type;
}

std::string Image::getTypeString() const
{
  std::string type_str;

  switch (this->type)
  {
    case Image::Type::bmp:
    {
      type_str = "bmp";
      break;
    }
    case Image::Type::png:
    {
      type_str = "png";
      break;
    }
    case Image::Type::jpg:
    {
      type_str = "jpg";
      break;
    }
  }

  return type_str;
}

void Image::setType(Image::Type type)
{
  this->type = type;
}

std::string const& Image::getCamera() const
{
  return this->camera;
}

void Image::setCamera(std::string const& camera)
{
  this->camera = camera;
}

std::string const& Image::getName() const
{
  return this->name;
}

void Image::setName(std::string const& name)
{
  this->name = name;
}

std::string const& Image::getCreationDate() const
{
  return this->creationDate;
}

void Image::setCreationDate(std::string const& creationDate)
{
  this->creationDate = creationDate;
}

std::string const& Image::getCreationTime() const
{
  return this->creationTime;
}

void Image::setCreationTime(std::string const& creationTime)
{
  this->creationTime = creationTime;
}

std::ostream& operator<<(std::ostream& os, Image const& image)
{
  os << "Image{type=" << image.getTypeString() <<
        ", camera=" << image.camera <<
        ", creationDate=" << image.creationDate <<
        ", creationTime=" << image.creationTime <<
        ", name=" << image.name <<
        "}";

  return os;
}
